tensorflow_datasets/core/features/video_feature.py

```diff
@@ -197,26 +185,80 @@ def to_json_content(self) -> Json:
         'ffmpeg_extra_args': self._extra_ffmpeg_args
     }
 
+  def _generate_video_html(self, images, framerate=24, encoding_format='mp4') -> str:
+    """Converts sequence of images into video string."""
+    imgs = len(str(len(images)))+1  # Find number of digits in len to give names.
+
+    with tempfile.TemporaryDirectory() as video_dir:
+      for i, img in enumerate(images):
+        f = os.path.join(video_dir, f'img{i:0{imgs}d}.png')
+        img.save(f, format='png')
+
+      ffmpeg_args = [self._ffmpeg_path, '-framerate', str(framerate), '-i',
+                    os.fspath(os.path.join(video_dir, f'img%0{imgs}d.png'))]
+
+      extra_ffmpeg_args = ['-vf', 'scale=128:128', '-vcodec', 'h264',
+                           '-pix_fmt', 'yuv420p', '-allow_sw', '1']
+      ffmpeg_stdin = None
+
+      output_pattern = os.path.join(video_dir, 'output.' + encoding_format)
+      ffmpeg_args += extra_ffmpeg_args
+      ffmpeg_args.append(output_pattern)
+      try:
+        _ffmpeg_run(ffmpeg_args, ffmpeg_stdin)
+
+        video = None
+        video_file = os.path.join(video_dir, 'output.mp4')
+        with tf.io.gfile.GFile(video_file, 'rb') as vf:
+          video = vf.read()
+
+        def write_buff(buff):
+          buff.write(video)
+        video_str = utils.get_base64(write_buff)
+        return (f'<video height="128" width="128" controls loop plays-inline>'
+                f'<source src="data:image/gif;base64,{video_str}"'
+                f' type="video/mp4" alt="Video"></video>')
+
+      except OSError:
+        # if ffmpeg is not installed, generate GIF using pngs
+        duration = (41*24)/framerate
+        def write_buff(buff):
+          images[0].save(
+              buff,
+              format='GIF',
+              save_all=True,
+              append_images=images[1:],
+              # Could add a frame_rate kwargs in __init__ to customize this.
+              duration=duration,  # 41ms / img ~= 24 img / sec
```

Comment:
Originally we picked 24 img/sec but this is too fast. Now the two FFMPEG and GIF framerate should match

Reply:
I have added a `framerate` parameter to the and am generating duration from that to keep comparable framerate between the two.

Sure now I will use the original relation to convert `framerate` (used by `mp4`) into `duration` in ms (used by `GIF`). 
```
duration = (1/framerate)*100   #framerate will be passed to the function, currently I'm using 10
```